pkg/collector/jobevent_watcher.go

```diff
@@ -218,6 +234,13 @@ func (e EventHandler) CheckJobIsWatched(jobNamespace string, jobName string) boo
 	return false
 }
 
+func (e EventHandler) CheckPodFilter(podName string) bool {
+	if podFilter == nil {
```

Comment:
Per comment above I personally don't love the global reference but it's up to you